runtime/container.go

```diff
@@ -508,10 +509,24 @@ func (c *container) createCmd(pid string, cmd *exec.Cmd, p *process) error {
 			close(p.cmdDoneCh)
 		}()
 	}()
-	if err := c.waitForCreate(p, cmd); err != nil {
+
+	ch := make(chan error)
+	go func() {
+		if err := c.waitForCreate(p, cmd); err != nil {
+			ch <- err
+			return
+		}
+		c.processes[pid] = p
```

Comment:
The routine that starts this one doesn't return until this one is done, so I think we should be fine